source/common/access_log/access_log_impl.cc

```diff
@@ -128,17 +128,16 @@ RuntimeFilter::RuntimeFilter(const envoy::config::accesslog::v3::RuntimeFilter&
       use_independent_randomness_(config.use_independent_randomness()) {}
 
 bool RuntimeFilter::evaluate(const StreamInfo::StreamInfo& stream_info,
-                             const Http::RequestHeaderMap& request_headers,
-                             const Http::ResponseHeaderMap&,
+                             const Http::RequestHeaderMap&, const Http::ResponseHeaderMap&,
                              const Http::ResponseTrailerMap&) const {
   // This code is verbose to avoid preallocating a random number that is not needed.
   uint64_t random_value;
   if (use_independent_randomness_) {
     random_value = random_.random();
-  } else if (stream_info.getRequestIDProvider() == nullptr) {
+  } else if (!stream_info.getStreamIdProvider().has_value()) {
     random_value = random_.random();
   } else {
-    const auto rid_to_integer = stream_info.getRequestIDProvider()->toInteger(request_headers);
+    const auto rid_to_integer = stream_info.getStreamIdProvider()->toInteger();
```

Comment:
It depends.

The stream id provider is created based on the header in the following snippet.
```
  // Set stream unique id to stream info after the request header has been generated or
  // overwritten (in the mutateRequestHeaders and mutateTracingRequestHeader).
  ASSERT(connection_manager_.config_.requestIDExtension() != nullptr);
  connection_manager_.config_.requestIDExtension()->setToStreamInfo(*request_headers_,
                                                                    filter_manager_.streamInfo());
```

So, except the http filters update the request id header, or the behavior will not be changed.

Reply:
And If we want a general stream id in the stream info, than we cannot depend the request headers directly.

Reply:
Or we can catch the request headers in the implementation of stream id provider. The lifetime of request headers shouldn't be problem because we always set request header pointer in the stream info anyway.

Reply:
I had tried to capture the request headers in the stream id provider's implementation. But I still a little worry about the lifetime of the captured request headers reference. orz

Reply:
@lizan New implementation was pushed. I did some refactor. In the new implementation, the `ActiveStream` instance will be captured in the stream id provider. This will ensure the provider will has no lifetime problems and the stream id of HTTP stream will be retrieved by the `RequestIDExtension` from the request headers dynamically.
So, there will no any behavior change after that.